R/expect_error_linter.R

```diff
@@ -0,0 +1,37 @@
+#' Require usage of expect_error(x) where appropriate
+#'
+#' If code is expected to produce an error, using `expect_that(x,
+#' throws_error())` works but is unadvised. It is better to use the function
+#' dedicated specifically for checking errors: `expect_error(x)`. The additional
+#' benefit is that you can also check for the expected error message using
+#' regular expressions.
+#'
+#' @evalRd rd_tags("expect_error_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+expect_error_linter <- function() {
+  # one case to match: expect_that(x, throws_error())
+  xpath <- glue::glue("//expr[
```

Comment:
Try to avoid `//expr` for performance reasons (cf. #1358)

Reply:
Also, e.g. in https://github.com/r-lib/lintr/blob/5cbc007df1032ec6d7b0d11c3abe7f692019166c/R/expect_comparison_linter.R#L15-L19 we exclude expressions from linting if the `info` argument is present. Is this necessary here, too?